distributed/tests/test_client.py

```diff
@@ -1116,6 +1117,27 @@ def g():
     result = yield y._result()
     assert result == 456
 
+@gen_cluster(client=True)
+def test_upload_file_zip(c, s, a, b):
```

Comment:
With `gen_cluster`, the workers should run in the current process, so they would inherit the current (process-wide) import state. For example, perhaps the `myfile` module seen in this test actually comes from another one of the tests here.

It seems all tests for `upload_file` are written in this style. I think it would deserve fixing.